systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectIsolatedST.java

```diff
@@ -594,6 +603,89 @@ void testSecretsWithKafkaConnectWithTlsAndScramShaAuthentication(ExtensionContex
+                    .endDockerOutput()
+                .endBuild()
+            .endSpec()
+            .build();
+
+        resourceManager.createResource(extensionContext, connect, ScraperTemplates.scraperPod(testStorage.getNamespaceName(), testStorage.getScraperName()).build());
+        LOGGER.info("Deploy NetworkPolicies for KafkaConnect");
+        NetworkPolicyResource.deployNetworkPolicyForResource(extensionContext, connect, KafkaConnectResources.deploymentName(testStorage.getClusterName()));
+
+        // How many messages should be sent and at what count should the test connector fail
+        int failMessageCount = 1;
+
+        Map<String, Object> echoSinkConfig = new HashMap<>();
+        echoSinkConfig.put("topics", testStorage.getTopicName());
+        echoSinkConfig.put("level", "INFO");
+        echoSinkConfig.put("fail.task.after.records", Integer.toString(failMessageCount));
+
+        LOGGER.info("Creating EchoSink connector");
+        resourceManager.createResource(extensionContext, KafkaConnectorTemplates.kafkaConnector(Constants.ECHO_SINK_CONNECTOR_NAME, testStorage.getClusterName())
+            .editOrNewSpec()
+                .withTasksMax(1)
+                .withClassName(Constants.ECHO_SINK_CLASS_NAME)
+                .withConfig(echoSinkConfig)
+                .withAutoRestart(new AutoRestartBuilder().withEnabled().build())
+            .endSpec()
+            .build());
+
+        // Send messages to topic
+        KafkaClients kafkaClients = new KafkaClientsBuilder()
+            .withTopicName(testStorage.getTopicName())
+            .withMessageCount(failMessageCount)
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
+            .withProducerName(testStorage.getProducerName())
+            .withNamespaceName(testStorage.getNamespaceName())
+            .build();
+
+        resourceManager.createResource(extensionContext, kafkaClients.producerStrimzi());
+        ClientUtils.waitForProducerClientSuccess(testStorage);
+        // Wait for echo-sink connector to pick up messages and fail tasks
+        KafkaConnectorUtils.waitForConnectorStatus(testStorage.getNamespaceName(), Constants.ECHO_SINK_CONNECTOR_NAME, NotReady);
+
+        KafkaConnectorStatus connectorState = KafkaConnectorResource.kafkaConnectorClient().inNamespace(testStorage.getNamespaceName()).withName(Constants.ECHO_SINK_CONNECTOR_NAME).get().getStatus();
+        String connectorTaskState = ((ArrayList<Map<String, String>>) connectorState.getConnectorStatus().get("tasks")).get(0).get("state");
+
+        // After task of echo-sink picks up messages from topic, it should fail.
+        assertEquals(connectorTaskState, "FAILED");
+        // Topic contains sent messages the whole test, so after the task auto-restarts,
+        // it still picks up messages and is supposed to fail again.
+        assertThat(connectorState.getAutoRestart().getCount(), is(not(0)));
```

Comment:
I think you can avoid it if you sent 5 messages instead of 1. That way it fails only once. I think that is important, because here, you should first wait for the `auto-restart counter being 1 (or >1 if needed). And then you should continue and wait for the auto-restart reset -> after another time (2 minutes if the restart was 1) it should reset the counter and the `.status.autoRestart` will be null.

Reply:
Thank you, added waiting for the roll, also increased messages to 5, this way as you said: echo-sink-connector reads-fails-restarts + after a few minutes autoRestartCount counter gets a reset.